ui/ui-components/components/session/SignIn.tsx

```diff
@@ -52,50 +107,133 @@ export default function SignInForm(props) {
     }
   };
 
-  async function getSetupSteps() {
+  const getSetupSteps = async () => {
     const { setupSteps }: Actions.SetupStepsList = await execApi(
       "get",
       `/setupSteps`
     );
     return { setupSteps };
-  }
+  };
+
+  useEffect(() => {
+    loadOauthOptions();
+    loadOAuthRequest();
+  }, []);
 
   return (
-    <Form id="form" onSubmit={handleSubmit(onSubmit)}>
-      <Form.Group>
-        <Form.Label>Email Address</Form.Label>
-        <Form.Control
-          disabled={loading}
-          autoFocus
-          required
-          name="email"
-          type="email"
-          placeholder="Email Address"
-          ref={register}
-        />
-        <Form.Control.Feedback type="invalid">
-          Email is required
-        </Form.Control.Feedback>
-      </Form.Group>
-
-      <Form.Group>
-        <Form.Label>Password</Form.Label>
-        <Form.Control
-          required
-          disabled={loading}
-          name="password"
-          type="password"
-          placeholder="Password"
-          ref={register}
-        />
-        <Form.Control.Feedback type="invalid">
-          A password is required
-        </Form.Control.Feedback>
-      </Form.Group>
-
-      <LoadingButton loading={loading} variant="primary" type="submit">
-        Sign In
-      </LoadingButton>
-    </Form>
+    <>
+      <Row>
+        <Col className="border-right">
+          <h2>Sign in with Email</h2>
+          <Form
+            id="form"
+            onSubmit={handleSubmit((data) => onSubmit(data, "password"))}
+          >
+            <Form.Group>
+              <Form.Label>Email Address</Form.Label>
+              <Form.Control
+                disabled={loading}
+                autoFocus
+                required
+                name="email"
+                type="email"
+                placeholder="Email Address"
+                ref={register}
+              />
+              <Form.Control.Feedback type="invalid">
+                Email is required
+              </Form.Control.Feedback>
+            </Form.Group>
+
+            <Form.Group>
+              <Form.Label>Password</Form.Label>
+              <Form.Control
+                required
+                disabled={loading}
+                name="password"
+                type="password"
+                placeholder="Password"
+                ref={register}
+              />
+              <Form.Control.Feedback type="invalid">
+                A password is required
+              </Form.Control.Feedback>
+            </Form.Group>
+
+            <LoadingButton loading={loading} variant="primary" type="submit">
```

Comment:
It would be best to separate the loading state between oAuth and others so that this button can be shown disabled while signing in with OAuth.

Reply:
Agree! Updated in #b86fe744c2979410cefa18a2242dbf1cedc16bdd